File: flutter_application_1/lib/models/crypto.dart
import 'dart:convert';


List<Crypto> cryptoFromJson(String str) =>
    List<Crypto>.from(json.decode(str).map((x) => Crypto.fromJson(x)));

class Crypto {
  final String id;
  final String symbol;
  final String name;
  final String? imageUrl;
  final double currentPrice;
  final int marketCapRank;
  final double priceChangePercentage24h;
  final double totalVolume;
  final double ath;
  final double athChangePercentage;

  Crypto({
    required this.id,
    required this.symbol,
    required this.name,
    required this.imageUrl,
    required this.currentPrice,
    required this.marketCapRank,
    required this.priceChangePercentage24h,
    required this.totalVolume,
    required this.ath,
    required this.athChangePercentage,
  });

  
  factory Crypto.fromJson(Map<String, dynamic> json) => Crypto(
        id: json["id"],
        symbol: json["symbol"],
        name: json["name"],
        imageUrl: json["image"],
        currentPrice: (json["current_price"] as num).toDouble(),
        marketCapRank: json["market_cap_rank"],
        priceChangePercentage24h:
            (json["price_change_percentage_24h"] as num).toDouble(),
        totalVolume: (json["total_volume"] as num? ?? 0).toDouble(),
        ath: (json["ath"] as num? ?? 0).toDouble(),
        athChangePercentage:
            (json["ath_change_percentage"] as num? ?? 0).toDouble(),
      );
}

File: go-api/models/crypto.go
package models

type CryptoData struct {
	ID                       string  `json:"id"`
	Symbol                   string  `json:"symbol"`
	Name                     string  `json:"name"`
	Image                    string  `json:"image"`
	CurrentPrice             float64 `json:"current_price"`
	MarketCapRank            int     `json:"market_cap_rank"`
	PriceChangePercentage24h float64 `json:"price_change_percentage_24h"`
	TotalVolume              float64 `json:"total_volume"`
	ATH                      float64 `json:"ath"`
	ATHChangePercentage      float64 `json:"ath_change_percentage"`
}
